Extension/src/LanguageServer/extension.ts

```diff
@@ -158,6 +158,9 @@ function sendActivationTelemetry(): void {
 }
 
 async function checkVsixCompatibility(): Promise<void> {
+    const ignoreMismatchedCompatibleVsix: PersistentState<boolean> = new PersistentState<boolean>("CPP.ignoreMismatchedCompatibleVsix", false);
```

Comment:
Do we have a facility to make this value persist only for the current installation?  I would want it to reappear the next time they install the extension (if it's wrong). Maybe we could append the version number to the state key which would be close enough?

Reply:
We could reset the persistent state after a non-mismatch install, right?

Reply:
I made the persistent var specific to the current version number.

@sean-mcmanus Currently, we will reset the persistent var such that 'it does not ignore' mismatches (so reports a warning if there is a mismatch - somewhat of a double-negative) unless there is a mismatch detected while already set to ignore it...  So, as soon as we see a non-mismatching (er, matching) install, future mismatches will trigger the warning.